src/nim-light-client/light_client_utils.nim

```diff
@@ -0,0 +1,879 @@
+    when update is SomeLightClientUpdateWithSyncCommittee:
+      not update.next_sync_committee_branch.isZeroMemory
+    else:
+      false
+  meta.has_finality =
+    when update is SomeLightClientUpdateWithFinality:
+      not update.finality_branch.isZeroMemory
+    else:
+      false
+  meta.num_active_participants =
+    countOnes(update.sync_aggregate.sync_committee_bits).uint64
+  meta
+
+func is_better_data*(new_meta, old_meta: LightClientUpdateMetadata): bool =
+  # Compare supermajority (> 2/3) sync committee participation
+  const max_active_participants = SYNC_COMMITTEE_SIZE.uint64
+  let
+    new_has_supermajority =
+      new_meta.num_active_participants * 3 >= max_active_participants * 2
+    old_has_supermajority =
+      old_meta.num_active_participants * 3 >= max_active_participants * 2
+  if new_has_supermajority != old_has_supermajority:
+    return new_has_supermajority > old_has_supermajority
+  if not new_has_supermajority:
+    if new_meta.num_active_participants != old_meta.num_active_participants:
+      return new_meta.num_active_participants > old_meta.num_active_participants
+
+  # Compare presence of relevant sync committee
+  let
+    new_has_relevant_sync_committee = new_meta.has_sync_committee and
+      new_meta.attested_slot.sync_committee_period() ==
+      new_meta.signature_slot.sync_committee_period
+    old_has_relevant_sync_committee = old_meta.has_sync_committee and
+      old_meta.attested_slot.sync_committee_period ==
+      old_meta.signature_slot.sync_committee_period
+  if new_has_relevant_sync_committee != old_has_relevant_sync_committee:
+    return new_has_relevant_sync_committee > old_has_relevant_sync_committee
+
+  # Compare indication of any finality
+  if new_meta.has_finality != old_meta.has_finality:
+    return new_meta.has_finality > old_meta.has_finality
+
+  # Compare sync committee finality
+  if new_meta.has_finality:
+    let
+      new_has_sync_committee_finality =
+        new_meta.finalized_slot.sync_committee_period ==
+        new_meta.attested_slot.sync_committee_period
+      old_has_sync_committee_finality =
+        old_meta.finalized_slot.sync_committee_period ==
+        old_meta.attested_slot.sync_committee_period
+    if new_has_sync_committee_finality != old_has_sync_committee_finality:
+      return new_has_sync_committee_finality > old_has_sync_committee_finality
+
+  # Tiebreaker 1: Sync committee participation beyond supermajority
+  if new_meta.num_active_participants != old_meta.num_active_participants:
+    return new_meta.num_active_participants > old_meta.num_active_participants
+
+  # Tiebreaker 2: Prefer older data (fewer changes to best data)
+  new_meta.attested_slot < old_meta.attested_slot
+
+
+template is_better_update*[A, B: SomeLightClientUpdate](
+    new_update: A, old_update: B): bool =
+  is_better_data(toMeta(new_update), toMeta(old_update))
+
+# Other helpers
+template assertLC*(cond: untyped, msg = "") =
+  assert(cond)
```

Comment:
Hrm, are you sure these asserts are compiled to anything? By default, `assert` in Nim is enabled only when compiling with `-d:debug`.

Reply:
That's right. Here is the fix 79d2b61b99c685e9ebc7c235990b2469403bc3f7